### PR TITLE
perf: Batch hypergraph updates

### PR DESCRIPTION
**Pull request by dmaretskyi**
### Summary
🔄🚀🧹

This pull request introduces a new way of handling signals in ECHO using multiple signal runtimes, such as Preact Signals or SolidJS. It simplifies the code and improves the performance and efficiency of the signal updates. It also removes the deprecated `registerSignalFactory` function and some unused imports and files. Additionally, it adds a new feature to the `SpaceListPanel` component to show the number of members in each space.

> _To make signals more efficient and fast_
> _We replaced some old files with new ones at last_
> _We used `compositeRuntime` to batch_
> _And removed some code that was a mismatch_
> _Now ECHO can handle multiple runtimes with a blast_

### Walkthrough
*  Register and use multiple signal runtimes for ECHO using the `compositeRuntime` object from `@dxos/echo-schema` ([link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-ea3963e9e0ef0e8fb4bff94dc35f8655e5f0f297acc6c55625ce53e072a35b3bR27), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-ea3963e9e0ef0e8fb4bff94dc35f8655e5f0f297acc6c55625ce53e072a35b3bL140-R160), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-9b3a1c6587b9d41bd1e1e60119c8921bc6a73f1cf399376070822062b2b5089dL20-R20), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-9b3a1c6587b9d41bd1e1e60119c8921bc6a73f1cf399376070822062b2b5089dL61-R61), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-874cfefd5ce6f29a1bbd0df27acce8a2aa03e20f12db8d444a09c6802006bc92L246-R246), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-874cfefd5ce6f29a1bbd0df27acce8a2aa03e20f12db8d444a09c6802006bc92L295-R295), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-874cfefd5ce6f29a1bbd0df27acce8a2aa03e20f12db8d444a09c6802006bc92L555-R555), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-786f42f6e60b94ab68fa74289bb879371a4b0cbca1f9af2a6279600e6c01b6efL12-R12), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-786f42f6e60b94ab68fa74289bb879371a4b0cbca1f9af2a6279600e6c01b6efL97-R97), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-786f42f6e60b94ab68fa74289bb879371a4b0cbca1f9af2a6279600e6c01b6efL108-R108), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-786f42f6e60b94ab68fa74289bb879371a4b0cbca1f9af2a6279600e6c01b6efL124-R124), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-786f42f6e60b94ab68fa74289bb879371a4b0cbca1f9af2a6279600e6c01b6efL130-R130), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-1af25d9108a1c6777d56dab1b0b904073fddf5b2a4f4a14e3f666405a7b92ffbL5-R83))
* Remove the import and usage of the deprecated `registerSignalFactory` function from `@dxos/echo-signals/react` in `ClientPlugin.tsx` and `ClientContext.tsx` ([link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-2980bfaacf44efa379f06240cc85cd206bc5b09d919e1ecfb8b170c6c4d08cfdL11), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-2980bfaacf44efa379f06240cc85cd206bc5b09d919e1ecfb8b170c6c4d08cfdL40), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-9a29f84a5795d1286ba51baf110e4f2d82247073f432fa10a53f3a39d80e30d8L13), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-9a29f84a5795d1286ba51baf110e4f2d82247073f432fa10a53f3a39d80e30d8L20), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-9a29f84a5795d1286ba51baf110e4f2d82247073f432fa10a53f3a39d80e30d8L80-L84), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-9a29f84a5795d1286ba51baf110e4f2d82247073f432fa10a53f3a39d80e30d8L102-R96))
* Register the signal runtime for ECHO using the `signal` and `batch` functions from `@preact/signals-core` and `@preact/signals-react` in `core.ts` and `react.ts` ([link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-22d084e419552304485e2a256c32f1aee7511125f3e567f79b1778960ddf16b8L5-R38), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-7dbacb84259891c4c4fd25efc621d7128f317bb925c2f46b852aa2a26f3fab5aL5-R38))
* Export a `registerSignalRuntime` function and a deprecated `registerSignalFactory` function for backward compatibility in `core.ts` and `react.ts` ([link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-22d084e419552304485e2a256c32f1aee7511125f3e567f79b1778960ddf16b8L5-R38), [link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-7dbacb84259891c4c4fd25efc621d7128f317bb925c2f46b852aa2a26f3fab5aL5-R38))
* Add the `"sideEffects": true` property to the `package.json` file of the `@dxos/echo-signals` package ([link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-ab890d883a6eed4dc05f2c87bd853f152774b7051ee7dcbafac4fa51ffacbecdR40))
* Delete the unused file `common.ts` from `@dxos/echo-signals` ([link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-cae17c0d60d522cdaa86b7ed6033ae50b751114f91882a421435f6cffcc80686))
* Add a new column to the `SpaceListPanel` component that displays the number of members in each space ([link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-bf5f3f312ecbe59ff02fb2ecb3634f997713e8ce53359fbed1ce8b32add2cc7dR43-R46))
* Update the year in the license header of the file `preact.ts` from 2022 to 2023 ([link](https://github.com/dxos/dxos/pull/4677/files?diff=unified&w=0#diff-2cf53e6a6c3fdc0d9491cb04cd215def560e405bc6b78fe8776890625f4382bdL2-R35))

### CODE REVIEW

File: packages/core/echo/echo-signals/src/common.ts [hunk 1]
```diff
@@ -1,24 +0,0 @@
-//
-// Copyright 2023 DXOS.org
-//
-
-import type { signal } from '@preact/signals-core';
-
-import { registerSignalFactory } from '@dxos/echo-schema';
-
-export const constructRegisterSignalFactory = (constructSignal: typeof signal) => () => {
```
Comment (wittjosiah) [inline packages/core/echo/echo-signals/src/common.ts:9]: Is there a reason you stopped having common code here?

File: packages/core/echo/echo-signals/package.json [hunk 2]
```diff
@@ -37,6 +37,7 @@
       ]
     }
   },
+  "sideEffects": true,
```
Comment (wittjosiah) [inline packages/core/echo/echo-signals/package.json:40]: what's this do?

File: packages/sdk/react-client/src/client/ClientContext.tsx [hunk 3]
```diff
@@ -99,15 +92,8 @@ export const ClientProvider = ({
   services: createServices,
   client: clientProvider,
   fallback: Fallback = () => null,
-  registerSignalFactory: register = true,
   onInitialized,
 }: ClientProviderProps) => {
-  useMemo(() => {
-    // TODO(wittjosiah): Ideally this should be imported asynchronosly because it is optional.
-    //   Unfortunately, aysnc import seemed to break signals React instrumentation.
-    register && registerSignalFactory();
```
Comment (wittjosiah) [inline packages/sdk/react-client/src/client/ClientContext.tsx:108]: How is this not breaking?

Comment (dmaretskyi) [reply packages/sdk/react-client/src/client/ClientContext.tsx:108]: not sure, but i didnt mean to actually change it -- reverted

File: packages/core/echo/echo-signals/package.json (hunk 2 above)
Comment (dmaretskyi) [reply packages/core/echo/echo-signals/package.json:40]: not relevant now, but it signals bundlers that a bare import has side effects:

```
// wont be removed during tree shaking
import '@dxos/echo-signals/react';
```

**Pull request merged**